File: src/mvc/cpp/item_delegate/main_window.cpp
#include "main_window.h"
#include "./ui_main_window.h"
#include "custom_delegate.h"
#include <QModelIndex>
#include <QStringListModel>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent), ui(new Ui::MainWindow) {
  ui->setupUi(this);

  auto model = new QStringListModel(this);
  model->setStringList(QStringList() << "Item 1"
                                     << "73"
                                     << "Item 3"
                                     << "Item 4"
                                     << "Item 5"
                                     << "Item 6"
                                     << "Item 7"
                                     << "Item 8"
                                     << "Item 9");
  ui->listView->setModel(model);
  ui->listView->setItemDelegate(new CustomDelegate(this));
}

MainWindow::~MainWindow() { delete ui; }

File: src/mvc/python/item_delegate/main_window.py
from custom_delegate import CustomDelegate
from PyQt6 import QtWidgets, uic
from PyQt6.QtCore import QStringListModel


class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        uic.loadUi("main_window.ui", self)

        model = QStringListModel(self)
        model.setStringList(
            [
                "Item 1",
                "74",
                "Item 3",
                "Item 4",
                "Item 5",
                "Item 6",
                "Item 7",
                "Item 8",
                "Item 9",
            ]
        )
        self.listView.setModel(model)
        self.listView.setItemDelegate(CustomDelegate(self))
        self.show()
